src/config.rs

```diff
@@ -119,24 +119,32 @@ pub struct LdapConfig {
 	/// Whether to update deleted entries
 	pub check_for_deleted_entries: bool,
 	/// TLS-related configuration
-	pub tls: LdapTlsConfig,
+	pub tls: Option<LdapTlsConfig>,
 }
 
 impl From<LdapConfig> for ldap_poller::Config {
 	fn from(cfg: LdapConfig) -> ldap_poller::Config {
+		let starttls = cfg.tls.clone().is_some_and(|tls| tls.danger_use_start_tls);
+		let no_tls_verify = cfg.tls.clone().is_some_and(|tls| tls.danger_disable_tls_verify);
+		let root_certificates_path = cfg.tls.clone().and_then(|tls| tls.server_certificate);
+		let client_key_path = cfg.tls.clone().and_then(|tls| tls.client_key);
+		let client_certificate_path = cfg.tls.clone().and_then(|tls| tls.client_certificate);
```

Comment:
Cloning `cfg` 5 times is not efficient, mb something like `cfg.tls.as_ref().and_then(|tls| tls.foo.clone())`? For bool fields there wouldn't be any need to clone as they are `Copy`

Reply:
Ah, hrm, yeah, makes sense, thanks!